lib/manager/gradle/index.js

```diff
@@ -178,8 +174,24 @@ async function executeGradle(config) {
   return true;
 }
 
-function getDockerRenovateGradleCommandLine(localDir) {
-  return `docker run --rm -v ${localDir}:${localDir} -w ${localDir} renovate/gradle ${GRADLE_DEPENDENCY_REPORT_COMMAND}`;
+function getGradleCommandLine(config) {
+  let cmd;
+  const gradlewExists = fs.exists('gradlew');
+  if (gradlewExists && !config.gitFs) {
+    logger.warn(
+      'Found gradle wrapper (gradlew) in the project, but --git-fs is not enabled, not using it.'
+    );
+  }
+  if (config.gitFs && gradlewExists) {
```

Comment:
shouldn't using docker take precedence? Meaning if I decide to use docker it should use it regardless of gradlew?

Reply:
As stated in #3135, the gradle wrapper (`gradlew`) should be privileged to system binary `gradle`. So I think it makes sense that, even when you want to use docker binaries, if the wrapper is found you use it. I'm not a lot familiar with gradle but I never seen a reason to use the system gradle when the wrapper was available.

Reply:
I think the use of docker is more related on how renovate infrastructure works, @rarkins  ?

Reply:
We are moving to an approach where you pick one of two options:

(a) Install the exact version of everything you need to use into PATH before running Renovate, or
(b) Renovate will detect the best version to run and then pull/execute that Docker image

It's important that this works in both situations. Now in the case of (b) we have this image: https://github.com/renovatebot/docker-gradle

How would it need to be updated to work in a gradlew scenario? We are not just installing gradle there but also adding some files.

@corecanarias line 18 in that Dockerfile looks wrong to me though, should it be deleted?

Reply:
For me binarySource should have precedence because it's a user configuration, meaning I explicitly said I want to use docker for this project, if I wanted to use gradlew I wouldn't put that argument in the configuration, right?

So the precedence should be:
* User overrides: by cli arguments or configuration
* gradlew if it exists
* gradle in the path

Reply:
OK, I'll modify that :+1: